guides/source/active_record_connection_management.md

```diff
@@ -0,0 +1,107 @@
+**DO NOT READ THIS FILE ON GITHUB, GUIDES ARE PUBLISHED ON http://guides.rubyonrails.org.**
+
+Active Record Connection Management
+===================================
+
+In general, Active Record will automagically manage database connections in an
+effecient manner. See [Configuring a Database](/configuring.html#configuring-a-database) and [Database pooling](configuring.html#database-pooling) for information about the basics of optimally configuring your application to use available resources.
+
+Sometimes, you will need to manage Active Record connections and/or connecton
+pools yourself. These situations are:
+
+* Configuring a multi-process web server
+* Spawning processes
+
+In previous version of rails, you needed to think about managing connections when
+spawning threads, but this is no longer the case.
+
+--------------------------------------------------------------------------------
+
+Configuring a multi-process web server
+--------------------------------------
+
+Each process in a multi-process web server needs its own database connection pool.
+If the server is also multi-threaded, the threads within each process will automatically
+share that processes' connection pool.
+
+Here is how to configure a server to set up its pools while booting your app:
+
+### Before forking
+
+Before the server forks, you need to disconnect the pool so that the forked processes
+don't get confused. Here's how to do that in a puma config:
```

Comment:
"get confused" is too vague - we should describe the problem(s) that are avoided by disconnecting the pool.